### DISCUSSION TITLE
New plugin documentation

### DISCUSSION THREAD

**Discussion by vnpraveendesu** [strongswan/strongswan#949] [Q&A]
Hi All,

I need to send SA details to my own module instead of Linux kernel. 

As I understood that this can be done using the Strongswan plugin framework.
Is there any similar kind of plugin available So that I can take it as a reference for my implementation.

Any kind of help will be appreciated.

Thanks in advance

Regards,
Praveen.

**Comment by strongX509** [comment]
If you want to write your own kernel-ipsec module you have implement the **kernel_ipsec** interface:

https://github.com/strongswan/strongswan/blob/master/src/libcharon/kernel/kernel_ipsec.h

One example is the **kernel-libipsec** plugin which implements the IPsec userland encryption:

https://github.com/strongswan/strongswan/tree/master/src/libcharon/plugins/kernel_libipsec
